sympy/polys/rootoftools.py

```diff
@@ -350,12 +351,14 @@ def __new__(cls, f, x, index=None, radicals=False, expand=True):
         if roots is not None:
             return roots[index]
 
+        if all((i.is_algebraic and (not i.is_symbol)) for i in poly.all_coeffs()):
```

Comment:
I'm not sure exactly what `is_symbol` means but `is_number` would be a better check here. If `is_number` is true then it means that the expression has no free symbols and can be evaluated to a complex number with `evalf`. Presumably this should also check `is_rational` as well although I haven't followed all the logic.